Enhancements

ojustino opened the pull request.

ojustino commented:
I realize there's an issue with the fix for #42 but I'm working on it along with #43.

astrofrog commented on pywwt/core.py:
```diff
@@ -197,10 +198,16 @@ def create_circle(self):
         return circle
 
-    def add_polygon(self):
+    def add_polygon(self,points=None):
```
Here maybe you can also add ``**kwargs`` and pass that to ``Polygon`` below so that the annotation traits can also be set on initialization?

astrofrog commented on pywwt/qt_widget.py:
```diff
@@ -127,7 +127,7 @@ def _run_js(self, js):
 class WWTQtWidget(BaseWWTWidget):
 
-    def __init__(self, block_until_ready=False):
+    def __init__(self, block_until_ready=False, size=None, **kwargs):
```
What is the ``size=None`` for?

ojustino replied:
I was following the code from the issue, which included `size` as an argument. I don't see it anywhere else so I'll take it out.

ojustino commented:
Since `**kwargs` allows for any keyword argument, do you think it'd be a good idea to use `[annotation].trait_names()` to warn users if they provide one that is not part of that annotation's class?

astrofrog commented:
@ojustino - this will need to be rebased now that I merged your other pull request. Yes maybe checking that the kwargs are in ``trait_names`` would be sensible.

astrofrog commented on pywwt/annotation.py:
```diff
@@ -27,19 +27,20 @@ class Annotation(HasTraits):
         # its own, we only want to react to changes in traits that have the wwt
         # metadata attribute (which indicates the name of the corresponding WWT
         # setting).
+        print('_an_otc',changed,sep=' ')
```
This should be removed

astrofrog commented on pywwt/annotation.py:
```diff
@@ -94,17 +95,23 @@ def _on_trait_change(self, changed):
+    
     fill = Bool(False, help='Whether or not the polygon should be filled').tag(wwt='fill', sync=True)
-    fill_color = Unicode('white', help='Assigns fill color for the polygon').tag(wwt='fillColor', sync=True)
+    fill_color = Unicode('blue', help='Assigns fill color for the polygon').tag(wwt='fillColor', sync=True)
```
I think this should be changed back to white

ojustino commented:
I wanted to let you see the results of the rebase. I'm working on the kwargs check, so please wait to merge if the branch is otherwise OK.

astrofrog commented:
@ojustino - this looks good. I think the reason the tests are failing is that i think the circle radius should be in degrees not arcseconds.

ojustino commented:
I changed the units, so we'll see how the checks go. I added the kwargs restriction to the Annotation class, too, so everything I planned to implement on this branch is now present.

astrofrog commented:
This looks great! You can ignore the AppVeyor CI failure for now, I'm working on that in a separate pull request.
